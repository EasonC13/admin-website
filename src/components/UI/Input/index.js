import PropTypes from "prop-types";
/** @jsx jsx */
import { css, jsx } from "@emotion/core";

const base = css`
  padding: 0.5rem 1rem;
  border: solid 1px black;
  outline: none;
`;

const rounded = css`
  border-radius: 5px;
`;

const pill = css`
  border-radius: 50px;
`;

const Input = ({ custom, variant, type, onChange, value, ...rest }) => (
  <input
    css={css(
      base,
      custom,
      variant === "rounded" && rounded,
      variant === "pill" && pill
    )}
    variant={variant}
    type={type}
    onChange={onChange}
    value={value}
    size={50}
    {...rest}
  />
);

export default Input;

Input.propTypes = {
  custom: PropTypes.object,
  className: PropTypes.string,
  style: PropTypes.object,
  rest: PropTypes.object,
  onChange: PropTypes.func,
  value: PropTypes.string,
  variant: PropTypes.string,
  type: PropTypes.string
};
