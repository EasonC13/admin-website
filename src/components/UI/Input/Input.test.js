// Input.spec.js
import React from "react";
import jest from "jest-mock";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import Input from ".";

describe("Input component", () => {
  it("should be defined", () => {
    expect(Input).toBeDefined();
  });

  it("shows Input text", () => {
    const input = renderer.create(<Input>Test Input</Input>);
    expect(input.toJSON()).toMatchSnapshot();
  });

  it("shows pill Input type", () => {
    const input = renderer.create(<Input type="pill">Test Input</Input>);
    const tree = input.toJSON();
    expect(tree.props.type).toContain("pill");
  });

  it("should call mock function when Input is changed", () => {
    enzyme.configure({ adapter: new Adapter() });
    const mockFn = jest.fn();
    const tree = enzyme.shallow(<Input onChange={mockFn}>Test Input</Input>);
    tree.simulate("change");
    expect(mockFn).toHaveBeenCalled();
  });
});
