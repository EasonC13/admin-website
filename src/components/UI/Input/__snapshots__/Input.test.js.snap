// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component shows Input text 1`] = `
<input
  className="css-1ku9ygt-Input"
  size={50}
>
  Test Input
</input>
`;
